pkg/virt-handler/container-disk/mount.go

```diff
@@ -45,9 +45,9 @@ type mounter struct {
 type Mounter interface {
 	ContainerDisksReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error)
 	MountAndVerify(vmi *v1.VirtualMachineInstance) (map[string]*containerdisk.DiskInfo, error)
-	MountKernelArtifacts(vmi *v1.VirtualMachineInstance, verify bool) error
+	mountKernelArtifacts(vmi *v1.VirtualMachineInstance, verify bool) error
```

Comment:
Can you please elaborate on the reason you lowered the case in these Interface function signatures?

Reply:
Of course.

Previously `MountAndVerify` and `MountKernelArtifacts` were two external functions that were assumed to be used from the outside at a particular order - on mount we need to first  `MountAndVerify()` and only then `MountKernelArtifacts()`, and when unmounting we need to first `UnmountKernelArtifacts()` then `Unmount()`.

For example, [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/vm.go#L2396) we can see virt-handler using them both.

This approach is both error prone and complicated since we're demanding an outside component to know all of these functions and the order they should be executed in. Instead, now mount / unmount of kernel artifacts is being done by `MountAndVerify()` and `Unmount()` automatically and are not exported ("private"). An external component does not have to "care" about these internal procedures which are done safely.

Reply:
After talking offline I understood @enp0s3 was referring to the fact that these "private methods" shouldn't be part of the public interface. This is absolutely right. Thanks for catching that! It's now removed from the interface.